src/main/java/dev/jbang/cli/Setup.java

```diff
@@ -4,35 +4,53 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.nio.file.StandardCopyOption;
 import java.nio.file.StandardOpenOption;
 
-import dev.jbang.ExitException;
 import dev.jbang.JdkManager;
 import dev.jbang.Settings;
-import dev.jbang.UnpackUtil;
 import dev.jbang.Util;
 
 import picocli.CommandLine;
 
 @CommandLine.Command(name = "setup", description = "Make jbang available for the user, either in the current session or permanently")
 public class Setup extends BaseCommand {
-	private static final String jbangUrl = "https://github.com/jbangdev/jbang/releases/latest/download/jbang.zip";
 
 	@CommandLine.Option(names = {
-			"--with-java" }, description = "Add Jbang's Java to the user's environment as well")
-	boolean withJava;
+			"--java" }, description = "Add Jbang's Java to the user's environment as well", negatable = true)
+	Boolean java;
 
 	@CommandLine.Option(names = {
-			"--fresh" }, description = "Force re-download and re-install of jbang")
-	boolean fresh;
+			"--force" }, description = "Force setup to be performed even when existing configuration has been detected")
+	boolean force;
 
 	@Override
 	public Integer doCall() throws IOException {
-		return setup(withJava, fresh);
+		boolean withJava;
+		if (java == null) {
+			withJava = guessWithJava();
+		} else {
+			withJava = java;
+		}
+		return setup(withJava, force);
+	}
+
+	public static boolean needsSetup() {
+		String envPath = System.getenv("PATH");
+		Path binDir = Settings.getConfigBinDir();
+		return !envPath.toLowerCase().contains(binDir.toString().toLowerCase());
+	}
+
+	public static boolean guessWithJava() {
```

Comment:
what does this do ? guessing if it needs java installed or how ?

Reply:
Indeed, if the user doesn't specify the option it will try to figure out if it should be installed or not. Basically if a JDK was installed by jbang and no JAVA_HOME is set it will assume it's okay to set up JAVA_HOME ourselves.

Reply:
Added javadoc

Reply:
what if java is available in the PATH from your system or some other package manager ?

this feels a  bit too magical to me...

Reply:
Jbang is all about magic :-)

If you have Java installed from other package managers you most likely have JAVA_HOME set, right? But still, we can easily add a check if java(c) is on the user's PATH and use that as an additional hint. I just want `jbang setup` to do the most useful thing for the user without them having to know about any extra options.

Reply:
no, neither brew, sdkman or rpm/yum installs sets a JAVA_HOME afaik. I wish they do but I don't think they do - exactly to avoid having two sources of "truth" conflicting (PATH vs JAVA_HOME).

jbangs magic of giving you a java if the one you have does not match //JAVA or no java available at all is fine as its pretty explicit/non-ambigious; but I think in this case there too many examples of JAVA_HOME not being set.

Reply:
I just tested it, but SDKMAN does set JAVA_HOME :-)